app/src/store/playground/playgroundStore.tsx

```diff
@@ -89,15 +89,13 @@ export const createPlaygroundStore = (
     operationType: "chat",
     inputMode: "manual",
     input: {
-      // TODO(apowell): When implementing variable forms, we should maintain a separate
-      // map of variableName to variableValue. This will allow us to "cache" variable values
-      // as the user types and will prevent data loss if users accidentally change the variable name
-      variables: {
-        // TODO(apowell): This is hardcoded based on the default chat template
-        // Instead we should calculate this based on the template on store creation
-        // Not a huge deal since this will be overridden on the first keystroke
+      // select variablesValueCache and variableKeys from the input in order
+      // to capture the current value of all valid variables across all instances' prompts
+      // if you just select variablesValueCache, then you may display stale values
+      variablesValueCache: {
```

Comment:
are we ever evicting anything from this cache?

Reply:
Nope. In theory it will evict when users navigate pages though.

We can add eviction easily if we think it will be a problem